syntax = "proto3";

package verification;

service TransactionVerificationService {
    rpc VerifyTransaction (VerifyTransactionRequest) returns (VerifyTransactionResponse);
}


message User {
    string name = 1;
    string contact = 2;
}

message CreditCard {
    string number = 1;
    string expirationDate = 2;
    string cvv = 3;
}

message Item {
  string name = 1;
  int32 quantity = 2;
}


message Transaction {
    repeated Item items = 1;
    User user = 2;
    CreditCard credit_card = 3;
    bool terms_and_conditions_accepted = 4;
}

message VerifyTransactionRequest {
    Transaction transaction = 1;
}

message VerifyTransactionResponse {
    bool is_valid = 1;
    string error_message = 2;
}